cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java

```diff
@@ -82,6 +82,27 @@ public static void createReflectConfig(Path nativeConfigPath, Package currentPac
             Map<String, String> testUtilityFunctions = testSuiteMap.get(moduleName).getTestUtilityFunctions();
             //Add init class
             if (testUtilityFunctions.containsKey(TEST_EXEC_FUNCTION)) {
+                ReflectConfigClass testTestExecuteGeneratedRefConfClz = new ReflectConfigClass(
+                        testSuiteMap.get(moduleName).getTestUtilityFunctions().get(TEST_EXEC_FUNCTION));
+                testTestExecuteGeneratedRefConfClz.addReflectConfigClassMethod(
+                        new ReflectConfigClassMethod(
+                                TEST_EXEC_FUNCTION,
+                                new String[]{
+                                        "io.ballerina.runtime.internal.scheduling.Strand",
+                                        "io.ballerina.runtime.api.values.BString",
+                                        "io.ballerina.runtime.api.values.BString",
+                                        "io.ballerina.runtime.api.values.BString",
+                                        "io.ballerina.runtime.api.values.BString",
+                                        "io.ballerina.runtime.api.values.BString",
+                                        "io.ballerina.runtime.api.values.BString",
+                                        "io.ballerina.runtime.api.values.BString",
+                                        "io.ballerina.runtime.api.values.BString",
+                                        "io.ballerina.runtime.api.values.BString",
+                                        "io.ballerina.runtime.api.values.BString"
+                                }
+                        )
+                );
+                classList.add(testTestExecuteGeneratedRefConfClz);
```

Comment:
Can we add a test for running mock test with native? May be we can do as a separate task

Reply:
Crated a separate task https://github.com/ballerina-platform/ballerina-lang/issues/40465

Reply:
we already have those mock tests https://github.com/ballerina-platform/ballerina-lang/blob/b30016edb09b37d029c10f404d16ebbd958097e8/cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestNativeImageCommandTest.java#L64..

Reply:
Can't we make them run with PR builds?